packages/loopring_v3/contracts/impl/libexchange/ExchangeAdmins.sol

```diff
@@ -96,27 +96,59 @@ library ExchangeAdmins
         );
     }
 
-    function purchaseDowntime(
+    function startOrContinueMaintenanceMode(
         ExchangeData.State storage S,
-        uint durationSeconds
+        uint durationMinutes
         )
         public
     {
         require(!S.isInWithdrawalMode(), "INVALID_MODE");
         require(!S.isShutdown(), "INVALID_MODE");
+        require(durationMinutes > 0, "INVALID_DURATION");
+
+        // Purchased downtime from a previous maintenance period or a previous call
+        // to startOrContinueMaintenanceMode can be re-used, so we need to calculate
+        // how many additional minutes we need to purchase
+        uint numMinutesLeft = S.getNumDowntimeMinutesLeft();
+        if (numMinutesLeft < durationMinutes) {
+            uint numMinutesToPurchase = durationMinutes.sub(numMinutesLeft);
+            uint costLRC = getDowntimeCostLRC(S, numMinutesToPurchase);
```

Comment:
can this be written as:
`uint costLRC = S.getDowntimeCostLRC(numMinutesToPurchase);`

Reply:
That doesn't seem possible because it's a function in the same library. If you do something like this: 

```
library ExchangeAdmins
{
    using ExchangeAdmins for ExchangeData.State;
    ...
}
```

then this strangely compiles, but cannot be deployed because it references itself.

Reply:
Fair enough.